scripts/host-check

```diff
@@ -796,31 +798,81 @@ def check_hugepages() -> CheckFuncReturn:
     )
 
 
+@dataclass(frozen=True)
+class PCIDriver:
+    """
+    Class to represent a PCI driver.
+    """
+
+    name: str
+
+    @property
+    def is_supported(self) -> bool:
```

Comment:
Arguably better for these to be actual methods rather than properties to indicate that some work is actually being done to determine this (and more common with the 'is_' prefix, as opposed to e.g. just `driver.supported`).